src/hotspot/share/runtime/flags/jvmFlagAccess.cpp

```diff
@@ -61,11 +61,15 @@ class TypedFlagAccessImpl : public FlagAccessImpl {
 
 public:
   JVMFlag::Error check_constraint_and_set(JVMFlag* flag, void* value_addr, JVMFlagOrigin origin, bool verbose) const {
+    verbose |= (origin == JVMFlagOrigin::ERGONOMIC);
     T value = *((T*)value_addr);
     const JVMTypedFlagLimit<T>* constraint = (const JVMTypedFlagLimit<T>*)JVMFlagLimit::get_constraint(flag);
     if (constraint != nullptr && constraint->phase() <= static_cast<int>(JVMFlagLimit::validating_phase())) {
       JVMFlag::Error err = typed_check_constraint(constraint->constraint_func(), value, verbose);
       if (err != JVMFlag::SUCCESS) {
+        if (origin == JVMFlagOrigin::ERGONOMIC) {
+          fatal("FLAG_SET_ERGO cannot be used to set an invalid value for %s", flag->name());
```

Comment:
Just occurred to me, should the `fatal` rather be a `vm_exit_during_initialization`?

Reply:
`vm_exit_during_initialization` is usually for conditions not controllable by the VM (out of memory, bad command-line options, etc). Here were have a programming error in the VM's ergo code. So `fatal()` is more appropriate. It also generates an hs_err file that can help debugging.

Reply:
Okay